code/game/objects/structures/mousemaze.dm

```diff
@@ -0,0 +1,175 @@
+#define NORTH 1
+#define SOUTH 2
+#define EAST 4
+#define WEST 8
+
+#define TOPLEFT NORTH | WEST
+#define TOPRIGHT (NORTH << SECONDARY) | EAST
+#define BOTTOMLEFT SOUTH | (WEST << SECONDARY)
+#define BOTTOMRIGHT (SOUTH << SECONDARY) | (EAST << SECONDARY)
+
+#define MAZEVIEWDIST 1
+
+#define EASTWEST 256
+#define NORTHSOUTH 1024
+
+#define DIRECTION_SECONDARY 4 //all secondary paths are bitwise shifted this value from their original direction
+#define CROSSING_SECONDARY 1
+//for example, North path two is 1<<4=16
+//exception: crossing at EASTWEST and NORTHSOUTH are shifted only once
+
+/obj/structure/mousemaze
+	name = "mouse labyrinth"
+	desc = "Used primarily in neurological studies to test the effects of brain damage."
+	icon = 'icons/obj/mousemaze.dmi'
+	icon_state = "mousemaze"
+	anchored = TRUE
+	var/mazeflags = 0 //Which paths are chiseled out?
+	var/datum/context_click/mazemaker/buildmaster
+	var/list/denizens = list() //This list associates mobs inside this mousemaze with which corner they're in. NW, NE, SW, SE
+
+/obj/structure/mousemaze/New()
+	..()
+	buildmaster = new(src)
+
+/obj/structure/mousemaze/Destroy()
+	for(var/atom/movable/AM in contents)
+		AM.forceMove(get_turf(src))
+	denizens.Cut()
+	..()
+
+/obj/structure/mousemaze/attackby(obj/item/weapon/W,mob/user, params)
+	if(istype(W,/obj/item/weapon/chisel))
+		buildmaster.action(W, user, params)
+	else if(istype(W,/obj/item/weapon/holder/animal))
+		var/obj/item/weapon/holder/animal/H = W
+		if(H.stored_mob.size == SIZE_TINY)
+			enter(H.stored_mob)
+	else if(iscrowbar(W))
+		visible_message("<span class='danger'>[user] begins dismantling the [src]!</span>")
+		if(do_after(user,src,count_set_bitflags(mazeflags) SECONDS + 1)) //1+paths set seconds to deconstruct.
```

Comment:
This will just set 0.1 seconds +`count_set_bitflags(mazeflags)` seconds, not `1 + count_set_bitflags(mazeflags)` seconds.

Reply:
Good catch